cpu/stm32_common/periph/flashpage.c

```diff
@@ -187,15 +188,16 @@ void flashpage_write_raw(void *target_addr, const void *data, size_t len)
 
     /* clear program bit again */
 #if defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F1) || \
-    defined(CPU_FAM_STM32L4)
+    defined(CPU_FAM_STM32L4) || defined(CPU_FAM_STM32F3)
     CNTRL_REG &= ~(FLASH_CR_PG);
 #endif
     DEBUG("[flashpage_raw] write: done writing data\n");
 
     /* lock the flash module again */
     _lock();
 
-#if defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F1)
+#if defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F1) || \
```

Comment:
This also need to be done for erase operations according to the reference manual p68.

> For program and erase operations on the Flash memory (write/erase), the internal RC
> oscillator (HSI) must be ON.